backend/src/services/booking.service.impl.ts

```diff
@@ -221,6 +221,24 @@ export default class BookingService implements IBookingService {
     return this.bookingRepository.countBookingsByStatus()
   }
 
+  async findByUserId(
+    userId: number,
+    hideInactive: boolean
+  ): Promise<Booking[]> {
+    logger.debug("findByUserId")
+
+    let allBookings = await this.bookingRepository.findByUser(userId)
+    if (hideInactive) {
+      const filteredBookings = allBookings.filter(
+        (booking) =>
+          booking.Status === BookingStatus.BOOKED ||
+          booking.Status === BookingStatus.DELIVERED
+      )
+      allBookings = filteredBookings
```

Comment:
isn't better to do this filter inside the sql query?

Reply:
I did it this way because this query always returns a small amount of data. But I agree with your suggestion. The change has been made. Thanks